### ISSUE TITLE
Segmentation fault on Tinycam connecting to Neolink - E1 pro

### ISSUE THREAD

**Issue by aleqcz** [thirtythreeforty/neolink#115]
**Describe the bug**
Segfault immediately after connecting Tinycam mobile app. VLC with same url works

**To Reproduce**
Steps to reproduce the behavior. Example:
1. Create this configuration file:
```
# A bind value of 0.0.0.0 means any network this computer can access
# You can chage this to a specfic network e.g. "192.168.1.101" here
# Or to no networks e.g. this computer only "127.0.0.1"
bind = "0.0.0.0"
# Default port is 8554 but you can change it by uncommenting the following
# bind_port = 8554

# Uncomment the following and supply a path to a valid PEM
# to activate TLS encryption.
# The PEM should contain the certificate and the private key
# If TLS is activated you must connect with "rtsps://" and not "rtsp://"
# certificate = "/path/to/pem/with/cert/and/key"

# You can password protect the rtsp server mount points by adding users
# like the following me and someone. If you do not add [[users]]
# then anyone can connect without a password or username
# To access such a stream try using a url such as "rtsp://me:mepass@192.168.1.101/driveway"

[[users]]
name = "admin"
pass = "..."

# [[users]]
# name = "someone"
# pass = "someonepass"

[[cameras]]
name = "e1"
username = "admin"
password = "..."
address = "192.168.1.82:9000"
# stream = "mainStream"
stream = "both"
format = "h264"
```

2. Launch Neolink (dockerized): docker-compose up
```version: '3.3'
services:
    neolink:
        ports:
            - '8554:8554'
        restart: always
        volumes:
            - '$PWD/config.toml:/etc/neolink.toml'
        image: thirtythreeforty/neolink
        environment:
            - GST_DEBUG=3
```

3. Connect Tinycam with following settings

**Log**
```~/docker/neolink>docker-compose up
Creating network "neolink_default" with the default driver
Creating neolink_neolink_1_9bcf0abc8634 ... done
Attaching to neolink_neolink_1_4762837141b5
neolink_1_4762837141b5 | [2020-12-26T00:20:24Z INFO  neolink] Neolink 0.3.0 (unknown commit) release
neolink_1_4762837141b5 | 0:00:00.526596433     7 0x55ff810f1af0 FIXME             vdpdecoder gstvdpdecoder.c:352:gst_vdp_decoder_class_init: Actually create srcpad template from hw capabilities
neolink_1_4762837141b5 | Error relocating /usr/lib/libp11-kit.so.0: reallocarray: symbol not found
neolink_1_4762837141b5 | Failed to load module: /usr/lib/gio/modules/libgiognutls.so
neolink_1_4762837141b5 | 0:00:00.585997868     7 0x55ff810f1af0 WARN      GST_PLUGIN_LOADING gstplugin.c:805:_priv_gst_plugin_load_file_for_registry: module_open failed: Error relocating /usr/lib/libp11-kit.so.0: reallocarray: symbol not found
neolink_1_4762837141b5 |
neolink_1_4762837141b5 | (gst-plugin-scanner:7): GStreamer-WARNING **: 00:20:25.162: Failed to load plugin '/usr/lib/gstreamer-1.0/libgstrtmp.so': Error relocating /usr/lib/libp11-kit.so.0: reallocarray: symbol not found
neolink_1_4762837141b5 | 0:00:00.637000706     7 0x55ff810f1af0 WARN      GST_PLUGIN_LOADING gstplugin.c:805:_priv_gst_plugin_load_file_for_registry: module_open failed: Error relocating /usr/lib/libp11-kit.so.0: reallocarray: symbol not found
neolink_1_4762837141b5 |
neolink_1_4762837141b5 | (gst-plugin-scanner:7): GStreamer-WARNING **: 00:20:25.213: Failed to load plugin '/usr/lib/gstreamer-1.0/libgstwebrtc.so': Error relocating /usr/lib/libp11-kit.so.0: reallocarray: symbol not found
neolink_1_4762837141b5 | 0:00:00.676861879     7 0x55ff810f1af0 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x120000: 'AVR (Audio Visual Research)' is not mapped
neolink_1_4762837141b5 | 0:00:00.676934253     7 0x55ff810f1af0 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x180000: 'CAF (Apple Core Audio File)' is not mapped
neolink_1_4762837141b5 | 0:00:00.676981313     7 0x55ff810f1af0 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x100000: 'HTK (HMM Tool Kit)' is not mapped
neolink_1_4762837141b5 | 0:00:00.677058670     7 0x55ff810f1af0 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0xc0000: 'MAT4 (GNU Octave 2.0 / Matlab 4.2)' is not mapped
neolink_1_4762837141b5 | 0:00:00.677109370     7 0x55ff810f1af0 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0xd0000: 'MAT5 (GNU Octave 2.1 / Matlab 5.0)' is not mapped
neolink_1_4762837141b5 | 0:00:00.677153502     7 0x55ff810f1af0 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x210000: 'MPC (Akai MPC 2k)' is not mapped
neolink_1_4762837141b5 | 0:00:00.677200802     7 0x55ff810f1af0 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0xe0000: 'PVF (Portable Voice Format)' is not mapped
neolink_1_4762837141b5 | 0:00:00.677249176     7 0x55ff810f1af0 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x160000: 'SD2 (Sound Designer II)' is not mapped
neolink_1_4762837141b5 | 0:00:00.677302672     7 0x55ff810f1af0 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x190000: 'WVE (Psion Series 3)' is not mapped
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z WARN  neolink] Without a server certificate, usernames and passwords will be exchanged in plaintext!
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z WARN  neolink] The format config option of the camera has been removed in favour of auto detection.
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z INFO  neolink] e1: Connecting to camera at 192.168.1.82:9000
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z INFO  neolink] e1: Connecting to camera at 192.168.1.82:9000
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z INFO  neolink] e1: Connected and logged in
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z INFO  neolink] e1: Starting video stream subStream
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z INFO  neolink] e1: Connected and logged in
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z INFO  neolink] e1: Camera time is already set: 2020-12-26 1:20:24 +1
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z INFO  neolink] e1: Camera reports firmware version v3.0.0.116_20110204
neolink_1_4762837141b5 | [2020-12-26T00:20:25Z INFO  neolink] e1: Starting video stream mainStream

neolink_1_4762837141b5 | 0:00:37.603347757     6 0x55ed804d3180 ERROR             rtspclient rtsp-client.c:1059:find_media: client 0x55ed804e41c0: not authorized for factory path /e1
neolink_1_4762837141b5 | 0:00:37.603416692     6 0x55ed804d3180 ERROR             rtspclient rtsp-client.c:3287:handle_describe_request: client 0x55ed804e41c0: no media
neolink_1_4762837141b5 | 0:00:37.625729207     6 0x55ed80382af0 WARN                  appsrc gstappsrc.c:1911:gst_app_src_push_internal:<vidsrc> do-timestamp=TRUE but buffers are provided before reaching the PLAYING state and having a clock. Timestamps will not be accurate!
neolink_1_4762837141b5 | 0:00:37.625825041     6 0x55ed80382af0 WARN                  appsrc gstappsrc.c:1911:gst_app_src_push_internal:<audsrc> do-timestamp=TRUE but buffers are provided before reaching the PLAYING state and having a clock. Timestamps will not be accurate!
neolink_1_4762837141b5 | 0:00:37.625979806     6 0x55ed80382af0 WARN                  appsrc gstappsrc.c:1911:gst_app_src_push_internal:<vidsrc> do-timestamp=TRUE but buffers are provided before reaching the PLAYING state and having a clock. Timestamps will not be accurate!
neolink_1_4762837141b5 | 0:00:37.634732865     6 0x55ed804ffb00 FIXME                default gstutils.c:4025:gst_pad_create_stream_id_internal:<audsrc:src> Creating random stream-id, consider implementing a deterministic way of creating a stream-id
neolink_1_4762837141b5 | 0:00:37.636141484     6 0x55ed805032a0 FIXME                default gstutils.c:4025:gst_pad_create_stream_id_internal:<vidsrc:src> Creating random stream-id, consider implementing a deterministic way of creating a stream-id
neolink_1_4762837141b5 | 0:00:37.638027289     6 0x55ed80503240 WARN               h264parse gsth264parse.c:1349:gst_h264_parse_handle_frame:<h264parse0> broken/invalid nal Type: 1 Slice, Size: 17464 will be dropped
neolink_1_4762837141b5 | 0:00:37.854372314     6 0x55ed80503240 WARN               h264parse gsth264parse.c:1349:gst_h264_parse_handle_frame:<h264parse0> broken/invalid nal Type: 1 Slice, Size: 18530 will be dropped
neolink_1_4762837141b5 | 0:00:37.943548225     6 0x55ed804d3180 FIXME              rtspmedia rtsp-media.c:4548:gst_rtsp_media_suspend: suspend for dynamic pipelines needs fixing
neolink_1_4762837141b5 | 0:00:37.948312733     6 0x55ed804d3180 FIXME              rtspmedia rtsp-media.c:4548:gst_rtsp_media_suspend: suspend for dynamic pipelines needs fixing
neolink_1_4762837141b5 | 0:00:37.948380606     6 0x55ed804d3180 WARN               rtspmedia rtsp-media.c:4587:gst_rtsp_media_suspend: media 0x55ed8053c1d0 was not prepared
neolink_1_4762837141b5 | 0:00:37.952195795     6 0x55ed804e6b60 FIXME                default gstutils.c:4025:gst_pad_create_stream_id_internal:<appsrc2:src> Creating random stream-id, consider implementing a deterministic way of creating a stream-id
neolink_1_4762837141b5 | 0:00:37.954247763     6 0x55ed804d3180 FIXME              rtspmedia rtsp-media.c:2867:gst_rtsp_media_seek_trickmode:<GstRTSPMedia@0x55ed8053c1d0> Handle going back to 0 for none live not seekable streams.
neolink_1_4762837141b5 | 0:00:37.955804669     6 0x55ed804ffb00 WARN                 basesrc gstbasesrc.c:3127:gst_base_src_loop:<audsrc> error: Internal data stream error.
neolink_1_4762837141b5 | 0:00:37.955876502     6 0x55ed804ffb00 WARN                 basesrc gstbasesrc.c:3127:gst_base_src_loop:<audsrc> error: streaming stopped, reason not-linked (-1)
neolink_1_4762837141b5 | 0:00:37.956174913     6 0x55ed804ffb00 WARN                   queue gstqueue.c:990:gst_queue_handle_sink_event:<queue1> error: Internal data stream error.
neolink_1_4762837141b5 | 0:00:37.957028467     6 0x55ed804ffb00 WARN                   queue gstqueue.c:990:gst_queue_handle_sink_event:<queue1> error: streaming stopped, reason not-linked (-1)
neolink_1_4762837141b5 | Segmentation fault (core dumped)
```

**Versions**
NVR software: Tinycam Pro
Neolink software: 3.0
Reolink camera model and firmware: E1 Pro v3.0.0.116_20110204

Many thanks for looking!

**Comment by QuantumEntangledAndy** [comment]
This is a known issue with docker. Please try to use the binaries instead.

Duplicate #46

**Comment by QuantumEntangledAndy** [comment]
@thirtythreeforty we have had a few duplicate issues about this docker segfault now. Should we add a note in  in the docker section of the readme about this know issue?

**Comment by QuantumEntangledAndy** [comment]
Closed as a duplicate of #46
